src/utstring.h

```diff
@@ -82,8 +82,8 @@ do {                                                       \
 #define utstring_new(s)                                    \
 do {                                                       \
    (s) = (UT_string*)malloc(sizeof(UT_string));            \
-   if (!(s)) oom();                                        \
-   utstring_init(s);                                       \
+   if ((s)) utstring_init(s);                              \
+   else oom();                                             \
```

Comment:
Why doesn't PVS-Studio also complain about the similar use of `oom` in `utstring_reserve`/`bincpy`/`concat`? Is it just that your particular usage pattern doesn't exercise that codepath in `utstring_reserve`?

I would expect PVS-Studio to know that `oom()` means `exit(-1)` which is a noreturn function.

Reply:
Oh, I forgot to mention an important information: it happens only if we disable the `oom`, e.g.:

```c
#ifndef oom
#define oom() (void) 0
#endif
```

Reply:
Ah. Well, if you disable `oom()`, then of course you have a bug, and PVS-Studio is right to warn you about it. The solution is to not do that.

Reply:
So, I wouldn't like to exit my application if oom... in this case, what do you recommend? :confused:

Reply:
It would be nice if the `utstring` could follow the same idea as `HASH_NONFATAL_OOM` from `uthash` to avoid to exit the application if OOM ...